client/post-editor/media-modal/index.jsx

```diff
@@ -209,15 +209,15 @@ module.exports = React.createClass( {
 		this.setView( ModalViews.IMAGE_EDITOR );
 	},
 
-	onImageEditorClose: function() {
+	onImageEditorClose: function( event, { hasEditedImage } = { hasEditedImage: false } ) {
```

Comment:
Thinking on this more, it's probably cleaner to pass through a second handler: `onImageEditorCancel`